nuxeo-features/nuxeo-features-test/src/main/java/org/nuxeo/jaxrs/test/JerseyClientHelper.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Antoine Taillefer <ataillefer@nuxeo.com>
+ */
+package org.nuxeo.jaxrs.test;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.ClientResponse;
+import com.sun.jersey.api.client.filter.HTTPBasicAuthFilter;
+import com.sun.jersey.client.apache4.ApacheHttpClient4Handler;
+import com.sun.jersey.core.header.InBoundHeaders;
+
+/**
+ * Helper for using the Jersey 1 {@link Client}.
+ *
+ * @since 9.3
+ */
+public final class JerseyClientHelper {
+
+    public static final int DEFAULT_CONNECTION_TIMEOUT = 60 * 1000; // 60 seconds
+
+    /**
+     * Default instance of a Jersey 1 {@link Client} relying on the Apache HTTP client,
+     * {@link #DEFAULT_CONNECTION_TIMEOUT} for the connection timeout parameters and no authentication.
+     */
+    public static final Client DEFAULT_CLIENT = clientBuilder().build();
+
+    private JerseyClientHelper() {
+        // Helper class
+    }
+
+    /**
+     * Allows to build a custom instance of a Jersey 1 {@link Client} relying on the Apache HTTP client.
+     * <p>
+     * For instance, you can set the connection timeout and credentials for basic authentication by calling:
+     *
+     * <pre>
+     * {@code
+     *     Client client = JerseyClientHelper.clientBuilder()
+     *                                       .setConnectTimeout(5000)
+     *                                       .setCredentials("joe", "password")
+     *                                       .build();
+     * }
+     *
+     * <pre>
+     */
+    public static ApacheHttpClientBuilder clientBuilder() {
+        return new ApacheHttpClientBuilder();
+    }
+
+    /**
+     * Makes the given {@link ClientResponse} {@link AutoCloseable} by wrapping it in a {@link CloseableClientResponse}.
+     */
+    public static CloseableClientResponse toAutoCloseable(ClientResponse response) {
+        return new CloseableClientResponse(response.getStatusInfo(), getResponseHeaders(response),
+                response.getEntityInputStream(), response.getClient().getMessageBodyWorkers());
```

Comment:
I would move this method to the `CloseableClientResponse` class and call it `of`.